src/storage/mod.rs

```diff
@@ -369,16 +387,60 @@ mod test {
     #[test]
     fn test_compression() {
         let orig = "fn main() {}";
-        let (data, alg) = compress(orig.as_bytes()).unwrap();
-        let blob = Blob {
-            mime: "text/rust".into(),
-            content: data.clone(),
-            path: "main.rs".into(),
-            date_updated: Utc::now(),
-            compression: Some(alg),
-        };
-        test_roundtrip(std::slice::from_ref(&blob));
-        assert_eq!(decompress(data.as_slice(), alg).unwrap(), orig.as_bytes());
+        for alg in CompressionAlgorithm::AVAILABLE {
+            println!("testing algorithm {}", alg);
+
+            let data = compress(orig.as_bytes(), *alg).unwrap();
+            let blob = Blob {
+                mime: "text/rust".into(),
+                content: data.clone(),
+                path: "main.rs".into(),
+                date_updated: Utc::now(),
+                compression: Some(*alg),
+            };
+            test_roundtrip(std::slice::from_ref(&blob));
+            assert_eq!(
+                decompress(data.as_slice(), *alg, std::usize::MAX).unwrap(),
+                orig.as_bytes()
+            );
+        }
+    }
+
+    #[test]
+    fn test_decompression_too_big() {
+        const MAX_SIZE: usize = 1024;
+
+        let small = &[b'A'; MAX_SIZE / 2] as &[u8];
+        let exact = &[b'A'; MAX_SIZE] as &[u8];
+        let big = &[b'A'; MAX_SIZE * 2] as &[u8];
+
+        for alg in CompressionAlgorithm::AVAILABLE {
+            let compressed_small = compress(small, *alg).unwrap();
+            let compressed_exact = compress(exact, *alg).unwrap();
+            let compressed_big = compress(big, *alg).unwrap();
```

Comment:
nit: you don't need all this `*alg`:
```suggestion
        for &alg in CompressionAlgorithm::AVAILABLE {
            let compressed_small = compress(small, alg).unwrap();
            let compressed_exact = compress(exact, alg).unwrap();
            let compressed_big = compress(big, alg).unwrap();
```